test/built-ins/TypedArray/prototype/copyWithin/coerced-values-end-detached-prototype.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019 Google. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.copywithin
+description: >
+  SECURITY: end argument is coerced to an integer values
+  causing array detachment, but the value is still defined
+  by a prototype
```

Comment:
this is wrong.

```
%TypedArray%.prototype.copyWithin ( target, start [ , end ] )

...
8. If end is undefined, let relativeEnd be len; else let relativeEnd be ? ToInteger(end).
9. If relativeEnd < 0, let final be max((len + relativeEnd), 0); else let final be min(relativeEnd, len).
10. Let count be min(final - from, len - to).
11. If count > 0, then
    a. NOTE: The copying must be performed in a manner that preserves the bit-level encoding of the source data.
    b. Let buffer be O.[[ViewedArrayBuffer]].
    c. If IsDetachedBuffer(buffer) is true, throw a TypeError exception.
```

Spec wise, there is always a check to validate the `this` value. 

In step 8, the TypedArray instance gets a detached buffer, and its prototype is changed, but the `TypedArray#copyWithin` function continues running. In step 11.c the method will throw the TypeError exception.

Reply:
I've added a better description, though this is still what the test does. It shouldn't be necessary, but this does cause some engines to misbehave despite the specification.